framework/helpers/BaseHtml.php

```diff
@@ -884,9 +884,12 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
             $name .= '[]';
         }
 
-        $formatter = isset($options['item']) ? $options['item'] : null;
-        $itemOptions = isset($options['itemOptions']) ? $options['itemOptions'] : [];
-        $encode = !isset($options['encode']) || $options['encode'];
+        $formatter = ArrayHelper::remove($options, 'item');
+        $itemOptions = ArrayHelper::remove($options, 'itemOptions', []);
+        $encode = ArrayHelper::remove($options, 'encode', true);
+        $separator = ArrayHelper::remove($options, 'separator', PHP_EOL);
```

Comment:
We have HTML output. Newline there is to format HTML nicely. `PHP_EOL` gives different output on Windows (`\r\n`) and Linux (`\n`) but HTML is viewed on many different machines. In theory it could break weird platform-specific text editors.

In practice nothing will seriously break: browsers are OK with any form of line endings. To me it's still kinda clueless to use `PHP_EOL` in this case but I'm not strictly against it anymore.